frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -385,30 +386,47 @@ export const cleanUpWorkload = (
   };
   if (isKnativeResource) {
     // delete knative resources
-    const knativeService = _.find(workload.resources.ksservices, { kind: 'Service' });
-    resourceData.metadata.name = _.get(knativeService, 'metadata.name', '');
-    batchDeleteRequests(knativeDeleteModels, resourceData);
+    switch (resource.kind) {
+      case EventSourceCronJobModel.kind:
+        deleteRequest(EventSourceCronJobModel, resource);
```

Comment:
@karthikjeeyar Do we really need an If..else here for knative/non-knative the switch..case depends only on the resource kind? Can we just use one switch..case here?

Reply:
@sahil143 if..else is added to identify and delete crd and non-crd based models in a generic way.